packages/debug/src/browser/editor/debug-hover-widget.ts

```diff
@@ -171,17 +175,43 @@ export class DebugHoverWidget extends SourceTreeWidget implements monaco.editor.
                 this.activate();
             }));
         }
-        if (!await this.hoverSource.evaluate(expression)) {
+        const expression = await this.hoverSource.evaluate(matchingExpression);
+        if (!expression) {
             toFocus.dispose();
             this.hide();
             return;
         }
-        this.editor.getControl().layoutContentWidget(this);
+
+        this.contentNode.hidden = false;
+        this.domNode.classList.remove();
+        ['number', 'boolean', 'string'].forEach(token => this.titleNode.classList.remove(token));
+        this.domNode.classList.remove('complex-value');
+        if (expression.hasElements) {
+            this.domNode.classList.add('complex-value');
+        } else {
+            this.contentNode.hidden = true;
+            if (expression.type === 'number' || expression.type === 'boolean' || expression.type === 'string') {
+                this.titleNode.classList.add(expression.type);
+            } else if (!isNaN(+expression.value)) {
+                this.titleNode.classList.add('number');
+            } else if (DebugVariable.booleanRegex.test(expression.value)) {
+                this.titleNode.classList.add('boolean');
+            } else if (DebugVariable.stringRegex.test(expression.value)) {
+                this.titleNode.classList.add('string');
+            }
+        }
+
+        super.show();
+        new Promise<void>(resolve => {
+            setTimeout(() => window.requestAnimationFrame(() => {
+                this.editor.getControl().layoutContentWidget(this);
+                resolve();
+            }), 0);
+        });
```

Comment:
Unawaited promise.

Reply:
Nice catch. I have fixed it. Please proceed with the review. Thanks!